### ISSUE TITLE
Normalization with number input feature does not occur with LudwigModel.evaluate() method

### ISSUE THREAD

**Issue by jimthompson5802** [ludwig-ai/ludwig#1911]
**Describe the bug**
When using the `LudwigModel.evaluate()` method, number input feature normalization does not occur.  This failure manifest itself with incorrect predicted values, i.e., incorrect predicted probabilities for a binary output feature.  This code illustrates the issue:

```
import os
import shutil

from ludwig.datasets import adult_census_income
from ludwig.evaluate import evaluate_cli


if __name__ == "__main__":
    # Loads the dataset as a pandas.DataFrame
    train_df, test_df, _ = adult_census_income.load(split=True, cache_dir="adult_census_income_data")

    import pandas as pd
    import numpy as np

    # setup pandas output options
    pd.set_option('display.max_colwidth', 50)
    pd.set_option("display.width", 120)
    pd.set_option("display.max_columns", 10)

    # make reproducible output sample
    np.random.seed(31)

    # Print sample of trainig and test data sets.
    print(f"Training Dataset(train_df):\n{train_df.sample(n=5)}")
    print(f"\nTest Dataset(test_df):\n{test_df.sample(n=5)}")

    import logging
    from ludwig.api import LudwigModel

    # define model configuration
    config = {'combiner': {'dropout': 0.2,
                  'num_fc_layers': 3,
                  'output_size': 128,
                  'type': 'concat'},
     'input_features': [{'name': 'age', 'type': 'number'},
                        {'name': 'workclass', 'type': 'category'},
                        {'name': 'fnlwgt', 'type': 'number'},
                        {'name': 'education', 'type': 'category'},
                        {'name': 'education-num', 'type': 'number'},
                        {'name': 'marital-status', 'type': 'category'},
                        {'name': 'occupation', 'type': 'category'},
                        {'name': 'relationship', 'type': 'category'},
                        {'name': 'race', 'type': 'category'},
                        {'name': 'sex', 'type': 'category'},
                        {'name': 'capital-gain', 'type': 'number'},
                        {'name': 'capital-loss', 'type': 'number'},
                        {'name': 'hours-per-week', 'type': 'number'},
                        {'name': 'native-country', 'type': 'category'}],
     'output_features': [{'name': 'income',
                          'num_fc_layers': 4,
                          'output_size': 32,
                          'preprocessing': {'fallback_true_label': ' >50K'},
                          'loss': {'type': 'binary_weighted_cross_entropy'},
                          'type': 'binary'}],
     'preprocessing': {'number': {'missing_value_strategy': 'fill_with_mean',
                                  'normalization': 'zscore'}},
     'trainer': {'epochs': 3, 'optimizer': {'type': 'adam'}}}


    # instantiate Ludwig model object
    model = LudwigModel(config=config, logging_level=logging.INFO)

    shutil.rmtree('results', ignore_errors=True)
    # Trains the model. This cell might take a few minutes.
    train_stats, preprocessed_data, output_directory = model.train(training_set=train_df,
                                                                   test_set=test_df)

    np.random.seed(13)
    eval_df = test_df.sample(n=1000)

    print(eval_df.head())

    # Generates predictions and performance statistics for the test set.
    shutil.rmtree('test_results', ignore_errors=True)
    print(f"\n>>>>>torch training flag before: {model.model.training}")
    print(f"\n>>>>PERFORMING api evalaute()<<<<<<<<<")
    test_stats, predictions, output_directory = model.evaluate(
      eval_df,
      collect_predictions=True,   # True
      collect_overall_stats=True, # True
      skip_save_eval_stats=False,   # False
      skip_save_predictions=False,  # False
      # skip_save_unprocessed_output=True,   # not specified
      output_directory="test_results",
      return_type="dict"
    )
    print(f"torch training flag after: {model.model.training}")

    pred_df = pd.read_parquet("test_results/predictions.parquet")
    print("\n>>>>> evaluate api from in-memory api trained model")
    print(pred_df.head())

```

Predicted probabilities are all 1.0 or 0.0.

```
  income_probabilities income_predictions  income_probabilities_ <=50K  income_probabilities_ >50K  income_probability
0           [1.0, 0.0]              <=50K                          1.0                         0.0                 1.0
1           [1.0, 0.0]              <=50K                          1.0                         0.0                 1.0
2           [1.0, 0.0]              <=50K                          1.0                         0.0                 1.0
3           [1.0, 0.0]              <=50K                          1.0                         0.0                 1.0
4           [1.0, 0.0]              <=50K                          1.0                         0.0                 1.0
```
